app/Svg/Generator.hs

```diff
@@ -45,9 +46,29 @@ buildSVG :: T.Text               -- ^ The name of the graph that is being built.
          -> Bool                 -- ^ Whether to include inline styles.
          -> IO ()
 buildSVG graphName courseMap filename styled = runDb $ do
-        gIds        :: [Key Graph]    <- selectKeysList [GraphTitle ==. graphName] []
-        let gId = if null gIds then toSqlKey 1 else head gIds
+        maybeGraph  :: Maybe (Entity Graph) <- selectFirst [GraphTitle ==. graphName] []
+        case fmap entityKey maybeGraph of
```

Comment:
This is better but still has a redundant layer of nesting: the "key" isn't actually needed any more, just the entity value.```